packages/matchbox/src/components/Drawer/Drawer.js

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import FocusLock from 'react-focus-lock';
+import { Transition } from 'react-transition-group';
+import { tokens } from '@sparkpost/design-tokens';
+import { Box } from '../Box';
+import { Portal } from '../Portal';
+import Header from './Header';
+import Content from './Content';
+import Footer from './Footer';
+import { useWindowEvent } from '../../hooks';
+import { onKey } from '../../helpers/keyEvents';
+import { secondsToMS } from '../../helpers/string';
+import { getChild } from '../../helpers/children';
+import { getRectFor } from '../../helpers/geometry';
+import { Overlay, Container } from './styles';
+
+function Drawer(props) {
+  const {
+    children,
+    closeOnEscape,
+    closeOnOutsideClick,
+    id,
+    onChange,
+    onClose,
+    open,
+    portalId,
+    position,
+  } = props;
+
+  const [footerHeight, setFooterHeight] = React.useState('0px');
+  const overlayRef = React.useRef();
+  const childrenRef = React.useRef();
+  const footerRef = React.useRef();
+
+  // Calls onClose when clicking outside drawer content
+  function handleOutsideClick(e) {
+    if (!closeOnOutsideClick) {
+      return;
+    }
+
+    const isOutside =
+      overlayRef.current &&
+      overlayRef.current.contains(e.target) &&
+      childrenRef.current &&
+      !childrenRef.current.contains(e.target);
+
+    if (isOutside && open && onClose) {
+      onClose();
+    }
+  }
+
+  // Calls onClose when receiving a escape keydown event
+  function handleEscape(e) {
+    if (closeOnEscape && open && onClose) {
+      onKey('escape', onClose)(e);
+    }
+  }
+
+  useWindowEvent('keydown', handleEscape);
+  useWindowEvent('click', handleOutsideClick);
+
+  React.useEffect(() => {
+    if (onChange) {
+      onChange(open);
+    }
+  }, [open]);
+
+  /**
+   * Drawer.Footer is a fixed element. This effect measures height of the footer and uses
+   * it as padding to offset Drawer.Content so its children isn't rendered underneath
+   * the footer
+   */
+  React.useLayoutEffect(() => {
+    if (!footerRef.current) {
+      setFooterHeight('0px');
+    } else {
+      setFooterHeight(`${getRectFor(footerRef.current).height}px`);
+    }
+  }, [footerRef.current, open]);
```

Comment:
You could potentially just use `position: sticky` here? Seems a little heavy to have JS handle this.

Reply:
Nice that works well! Theres a difference in scrollbar height but thats okay